### ISSUE TITLE
Unable to align Button content without sx prop css selectors

### ISSUE THREAD

**Issue by sdehm** [primer/react#4115]
### Description

The `Button` component has an `alignContent` prop which allows aligning the text of the button for cases such as with `block` enabled. This is shown in the storybook `Button` playground like [this](https://primer.style/react/storybook/?path=/story/components-button--playground&args=block:!true;alignContent:start).

When this prop is used however there is an eslint warning `primer-react/no-system-props` with more info [here](https://github.com/primer/eslint-plugin-primer-react/blob/main/docs/rules/no-system-props.md) that recommends against using "styled-system props" such as `alignContent`. While it is true that you can use the sx prop to align the content by styling the nested spans, like the example below, this isn't immediately clear.

```js
sx={{
  "& span": {
    justifyContent: "start",
  },
}}
```

### Steps to reproduce

Create a `Button` component with `block` and `alignContent="start"` and observe the eslint warning.

### Version

v36.4.0

**Comment by pksjce** [comment]
The removal of the system properties was AFAIK [intentional](https://github.com/primer/eslint-plugin-primer-react/blob/main/docs/rules/no-system-props.md). We don't have `alignContent` listed in our [documentation](https://primer.style/react/Button). However we missed removing it from the storybooks. I don't see how else to achieve this without targeted css.
I'd improve the target by doing

```js
sx={{
  "data-component:'text": {
    justifyContent: "start",
  },
}}
```

Not sure if there are other solutions.

**Comment by pksjce** [comment]
Further on we could think of adding `blockAlignContent` or some such prop if this is such a common scenario?

**Comment by siddharthkp** [comment]
new eslint-plugin-primer-react version released: [v4.1.0](https://github.com/primer/eslint-plugin-primer-react/releases/tag/v4.1.0), you can [upgrade the version in dotcom](https://github.com/github/github/blob/master/npm-workspaces/react-core/package.json#L7) now
